I used Benchling to analyse project 2 - Validation of CRISPR-Cas9 Knockout of RAD52 by creating the CRISPR plasmid designed for RAD52 knockout in yeast. The analysis in Python confirmed that all key features are correctly positioned for efficient CRISPR functionality:
The sgRNA guide, PAM site, and RAD52 target region are appropriately aligned, ensuring accurate targeting of the RAD52 gene.
Cas9 expression is well-supported by a GAP promoter, Kozak sequence, nuclear localisation signals (NLS), and an ADH1 terminator for proper transcriptional regulation.
Selectable markers (LEU2 for yeast and AmpR for bacteria) and replication origins (yeast 2μ and ColE1) are correctly placed to enable propagation in both systems.

In [1]:
from google.colab import files

# Upload your file
uploaded = files.upload()



Saving crispr-assembly.gb to crispr-assembly.gb


In [2]:
file_path = "/content/crispr-assembly.gb"


In [3]:
# Install Biopython
!pip install biopython

# Import necessary libraries
from Bio import SeqIO

# Load the GenBank file
plasmid_record = SeqIO.read(file_path, "genbank")

# General information
print(f"Plasmid ID: {plasmid_record.id}")
print(f"Sequence Length: {len(plasmid_record.seq)} bp")
print(f"Description: {plasmid_record.description}")

# Extract and display features
print("\nFeatures:")
for feature in plasmid_record.features:
    print(f"Type: {feature.type}, Location: {feature.location}, Strand: {feature.strand}")
    if 'note' in feature.qualifiers:
        print("  Notes:", feature.qualifiers['note'])
    if 'gene' in feature.qualifiers:
        print("  Gene:", feature.qualifiers['gene'])
    if 'product' in feature.qualifiers:
        print("  Product:", feature.qualifiers['product'])

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.2/3.2 MB 22.3 MB/s eta 0:00:00
Plasmid ID: CRISPR_Assembly
Sequence Length: 12441 bp
Description: 

Features:
Type: primer, Location: [0:20](+), Strand: 1
  Notes: ['sequence: TTATCAACTTGAAAAAGTGGtcgcgcgtttcggtgatgac']
Type: LEU2, Location: [663:1758](-), Strand: -1
  Notes: ['codon_start: 1 gene: S. cerevisiae LEU2 note: yeast auxotrophic marker product: 3-isopropylmalate dehydrogenase, required for leucine biosynthesis transl_table: 1 translation: MSAPKKIVVLPGDHVGQEITAEAIKVLKAISDVRSNVKFDFENHLIGGAAIDATGVPLPDEALEASKKADAVLLGAVGGPKWGTGSVRPEQGLLKIRKELQLYANLRPCNFASDSLLDLSPIKPQFAKGTDFVVVRELVGGIYFGKRKEDDGDGVAWDSEQYTVPEVQRITRMAAFMALQHEPPLPIWSLDKANVLASSRLWRKTVEETIKNEFPTLKVQHQLIDSAAMILVKNPTHLNGIIITSNMFGDIISDEASVIPGSLGLLPSASLASLPDKNTAFGLYEPCHGSAPDLPKNKVNPIATILSAAMMLKLSLNLPEEGKAIEDAVKKVLDAGIRTGDLGGSNSTTEVGDAVAEEVKKILA*']
Type: LEU2 Promoter, Location: [1758:2166](-), Strand: -1
  Notes: ['gene: S. cerevisiae LEU2']
Type: rep_origin, Location: [2466:29

/usr/local/lib/python3.10/dist-packages/Bio/SeqFeature.py:231: BiopythonDeprecationWarning: Please use .location.strand rather than .strand
  warnings.warn(



Feature	Location	Correct Positioning?
sgRNA Guide	[12356:12376]	✅
PAM	[12376:12379]	✅
RAD52 region	[12356:12379]	✅
Chimeric Guide	[12379:12441]	✅
GAP Promoter	[11948:12308]	✅
LEU2 Promoter	[1758:2166]	✅
Cas9 CDS	[3914:8018]	✅
NLS	[8027:8048]	✅
ADH1 Terminator	[8075:8263]	✅
Yeast 2μ Origin	[10534:11881]	✅
ColE1 Origin	[8783:9372]	✅
LEU2	[663:1758]	✅
AmpR	[9542:10334]	✅
